opentelemetry-api/src/opentelemetry/distributedcontext/__init__.py

```diff
@@ -11,3 +11,113 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+from contextlib import contextmanager
+import typing
+
+PRINTABLE = set(chr(num) for num in range(32, 127))
+
+
+class EntryMetadata:
+    """A class representing metadata of a DistributedContext entry
+
+    Args:
+        entry_ttl: The time to live (in service hops) of an entry. Must be
+                   initially set to either :attr:`EntryMetadata.NO_PROPAGATION`
+                   or :attr:`EntryMetadata.UNLIMITED_PROPAGATION`.
+    """
+
+    NO_PROPAGATION = 0
+    UNLIMITED_PROPAGATION = -1
+
+    def __init__(self, entry_ttl: int) -> None:
+        self.entry_ttl = entry_ttl
+
+
+class EntryKey(str):
+    """A class representing a key for a DistributedContext entry"""
+
+    def __new__(cls, value: str) -> "EntryKey":
+        return cls.create(value)
+
+    @staticmethod
+    def create(value: str) -> "EntryKey":
+        if len(value) > 255 or any(c not in PRINTABLE for c in value):
+            raise ValueError("Invalid EntryKey", value)
+
+        return typing.cast(EntryKey, value)
+
+
+class EntryValue(str):
+    """A class representing the value of a DistributedContext entry"""
+
+    def __new__(cls, value: str) -> "EntryValue":
+        return cls.create(value)
+
+    @staticmethod
+    def create(value: str) -> "EntryValue":
+        if any(c not in PRINTABLE for c in value):
+            raise ValueError("Invalid EntryValue", value)
+
+        return typing.cast(EntryValue, value)
```

Comment:
[`typing.cast`](https://docs.python.org/3/library/typing.html#typing.cast) is a no-op at runtime. Thus, this actually returns a `str` but makes mypy believe that it is actually a `EntryValue`. I think you want this:

```suggestion
        return EntryValue(value)
```

Reply:
Could this be done by using https://docs.python.org/3/library/typing.html#newtype instead?

Reply:
OK, you convinced me 👍 Even though I don't like that usage of cast, but maybe that's just my C++ background and in Python this is actually idiomatic.